src/figuresClasses.ts

```diff
@@ -1,11 +1,87 @@
-export interface Figure {}
+export interface Figure {
+  shape: `triangle` | `circle` | `rectangle`;
+  color: `red` | `green` | `blue`;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('the side should be bigger than 0');
+    }
+
+    if (!this.isValidTriangle(a, b, c)) {
+      throw new Error('a longest side should be bigger than sum of two others');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const result = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(result * 100) / 100;
+  }
```

Comment:
According to the checklist, the 'getArea' function should not accept any parameters. All necessary values are already available as object fields. Please remove the parameters from the 'isValidTriangle' method call within 'getArea'.